src/Carbon/Traits/Serialization.php

```diff
@@ -127,9 +128,14 @@ public function __sleep()
      */
     public function __wakeup()
     {
-        if (get_parent_class() && method_exists(parent::class, '__wakeup')) {
+        // FatalError occurs when calling __wakeup method in PHP 7.4 or later.
+        // @codeCoverageIgnoreStart
+        if (version_compare(PHP_VERSION, '7.4.0-dev', '>=')) {
+            parent::__construct($this->{$this->dumpProperties[0]}, $this->timezone !== null ? new DateTimeZone($this->timezone) : null);
+        } elseif (get_parent_class() && method_exists(parent::class, '__wakeup')) {
             parent::__wakeup();
         }
+        // @codeCoverageIgnoreEnd
```

Comment:
We have multiple little drawbacks with this.

`DateTime` has an explicitly [documented `__wakeup` method](https://www.php.net/manual/en/datetime.wakeup): so we should keep it to be future-proof and consistent with parent class.

The condition is too strict. [As per documentation](https://www.php.net/manual/en/language.oop5.magic.php),  `__sleep()` and `__wakeup()` are for `serialize()` and `unserialize()` and `unserialize(serialize(Carbon::parse('2018-06-01')))` works with no issue. Work-around should be at least scoped to `extension_loaded('msgpack')` and at best called only when calling `msgpack_unpack()` but not when calling `unserialize()`

Last (but that's not expected to be a standard usage, this will be a BC for usages of the trait like this:

```php
class Foo
{
    public $date = '2021-07-07 12:00:00';
    public $timezone_type = 3;
    public $timezone = 'America/Toronto';

    public function __construct()
    {
    }

    public function __wakeup()
    {
        echo "Hello";
    }
}

class Bar extends Foo
{
    use \Carbon\Traits\Serialization;
}
```
Output "Hello" before this PR, will output nothing after.
And worse if we don't have a `__construct` method:
Output "Hello" before this PR, will throw a fatal exception after.